### PR TITLE
heap sort in C

### LINKED ISSUE #23: [Algo]: Add Heapsort in C

### PR DESCRIPTION
**Pull request by PinkiePie1189**
**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

Fixes #23 

**By submitting this pull request I confirm I've read and complied with the
below declarations.**

- [x] I have read the [Contribution guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [commit guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#write-good-commit-messages) for this project.
- [x] My code follows the [standard code structure](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#samples-code).
- [x] This pull request has a descriptive title. For example, `{Tag}: Add {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.
- [x] This pull request shall only be reviewed and merged once all the checks passes. No maintainer or supporter shall be obliged to review it before this condition is met.
- [x] I have mentioned the issue number correctly (with hyperlink) in this pull request description.

After you submit your pull request, **DO NOT** click the 'Update Branch' button.

### CODE REVIEW

File: heap_sort/c/heap.c [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+			} else {
+				son = 2*node;
+			}
+		} else if (2*node <= h->N) {
+			son = 2*node;
+		}
+
+		if(h->heap[node] < h->heap[son]) {
+			son = 0;
+		}
```
Comment (sangamcse) [inline heap_sort/c/heap.c]: Missing space before ( in while(  [whitespace/parens] [5]

*Origin: CPPLintBear, Section: `all.cpplint`.*

**Comment by sangamcse** [comment]
Comment on 17058c10df354325cd282837d0d75d566c191b7f, file heap_sort/c/heapsort.c, line 12.

Missing spaces around =  [whitespace/operators] [4]

*Origin: CPPLintBear, Section: `all.cpplint`.*

**Comment by sangamcse** [comment]
Comment on 17058c10df354325cd282837d0d75d566c191b7f, file heap_sort/c/heapsort.c, line 18.

Missing spaces around =  [whitespace/operators] [4]

*Origin: CPPLintBear, Section: `all.cpplint`.*

**Comment by sangamcse** [comment]
Comment on 17058c10df354325cd282837d0d75d566c191b7f, file heap_sort/c/heapsort.c, line 28.

Missing spaces around =  [whitespace/operators] [4]

*Origin: CPPLintBear, Section: `all.cpplint`.*

**Comment by sangamcse** [comment]
Comment on 17058c10df354325cd282837d0d75d566c191b7f, file heap_sort/c/heapsort.c, line 32.

Missing spaces around =  [whitespace/operators] [4]

*Origin: CPPLintBear, Section: `all.cpplint`.*

**Comment by sangamcse** [comment]
Comment on 3a10b97377fd537989312203e842ccd3727873d9, file heap_sort/c/heapsort.c, line 32.

Missing spaces around =  [whitespace/operators] [4]

*Origin: CPPLintBear, Section: `all.cpplint`.*

**Comment by sangamcse** [comment]
Comment on 3a10b97377fd537989312203e842ccd3727873d9, file heap_sort/c/heap.c, line 43.

Missing space before ( in while(  [whitespace/parens] [5]

*Origin: CPPLintBear, Section: `all.cpplint`.*

**Comment by PinkiePie1189** [comment]
can you please explain to me why the travis CI build is failing?

File: heap_sort/c/heap.c [hunk 2]
```diff
@@ -0,0 +1,61 @@
+#include "./heap.h"
+
```
Comment (sangamcse) [inline heap_sort/c/heap.c]: heap_sort/c/heap.c should include its header file heap_sort/c/heap.h  [build/include] [5]

*Origin: CPPLintBear, Section: `all.cpplint`.*

**Comment by PinkiePie1189** [comment]
@sangamcse can you please explain why my PR was closed. I'm slightly confused

**Pull request closed without merging**